assembly/src/ast/parsers/constants.rs

```diff
@@ -0,0 +1,326 @@
+use super::{Felt, LocalConstMap, ParsingError, StarkField, String, Token, Vec};
+use core::fmt::Display;
+
+// CONSTANT VALUE EXPRESSIONS
+// ================================================================================================
+
+/// An operation used in constant expressions
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum Operation {
+    Add,
+    Sub,
+    Mul,
+    FeltDiv,
+    IntDiv,
+    LPar,
+    RPar,
+    Value(Felt),
+}
+
+impl Display for Operation {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        use Operation::*;
+
+        match self {
+            Add => write!(f, "+"),
+            Sub => write!(f, "-"),
+            Mul => write!(f, "*"),
+            FeltDiv => write!(f, "/"),
+            IntDiv => write!(f, "//"),
+            LPar => write!(f, "("),
+            RPar => write!(f, ")"),
+            Value(v) => write!(f, "{}", v),
+        }
+    }
+}
+
+/// Calculates expression in the constant value.
+pub fn calculate_const_value(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+) -> Result<Felt, ParsingError> {
+    let postfix_expression = build_postfix_expression(op, expression, constants)?;
+    evaluate_postfix_expression(op, expression, postfix_expression)
+}
+
+/// Parses constant value expression and transforms it into the postfix notation
+fn build_postfix_expression(
+    op: &Token,
+    expression: &str,
+    constants: &LocalConstMap,
+) -> Result<Vec<Operation>, ParsingError> {
+    let mut stack = Vec::new();
+    let mut postfix_expression = Vec::new();
+
+    let mut operation_iterator =
+        OperationIterator::new(op, expression, ['+', '-', '*', '/', '(', ')'], constants);
+
+    let mut operation = operation_iterator.next()?;
+
+    while operation.is_some() {
```

Comment:
I think you should be able to do something like this here:
```Rust
while let Some(operation) = operation_iterator.next()? {
```
If this works, you can also get rid of the line 99 below.